src/test/java/uk/gov/ons/ctp/integration/rhsvc/message/impl/CaseEventReceiverImplUnit_Test.java

```diff
@@ -1,122 +1,67 @@
-    when(mockRespondentDataRepo.readSurvey(any())).thenReturn(Optional.of(surveyUpdate));
-    caseEvent.setHeader(header);
     target.acceptCaseEvent(caseEvent);
+
     verify(mockRespondentDataRepo, times(0))
         .writeCaseUpdate(caseEvent.getPayload().getCaseUpdate());
   }
 
   @Test
-  public void test_acceptCaseEvent_invalidCase() throws Exception {
-    Header header = new Header();
-    header.setMessageId(UUID.fromString("c45de4dc-3c3b-11e9-b210-d663bd873d93"));
+  public void testExceptionThrown() throws CTPException {
     CaseEvent caseEvent = FixtureHelper.loadPackageFixtures(CaseEvent[].class).get(0);
-    caseEvent.setHeader(header);
-    SurveyUpdate surveyUpdate = new SurveyUpdate();
-    surveyUpdate.setSurveyId("c45de4dc-3c3b-11e9-b210-d663bd873d93");
-    surveyUpdate.setSampleDefinitionUrl("test/social.json");
-    CollectionExercise collectionExercise = new CollectionExercise();
-    collectionExercise.setCollectionExerciseId(
-        caseEvent.getPayload().getCaseUpdate().getCollectionExerciseId());
-    when(mockRespondentDataRepo.readCollectionExercise(any()))
-        .thenReturn(Optional.of(collectionExercise));
-    when(mockRespondentDataRepo.readSurvey(any())).thenReturn(Optional.of(surveyUpdate));
-    Mockito.doThrow(new CTPException(CTPException.Fault.RESOURCE_NOT_FOUND))
-        .when(mockRespondentDataRepo)
-        .writeCaseUpdate(any());
-    assertThrows(CTPException.class, () -> target.acceptCaseEvent(caseEvent));
-  }
+    when(acceptableEventFilter.filterAcceptedEvents(any(), any(), any(), any())).thenReturn(true);
+    doThrow(new CTPException(CTPException.Fault.SYSTEM_ERROR)).when(mockRespondentDataRepo).writeCaseUpdate(any());
 
-  @Test
-  public void test_acceptCaseEvent_missingSurvey() throws Exception {
-    Header header = new Header();
-    header.setMessageId(UUID.fromString("c45de4dc-3c3b-11e9-b210-d663bd873d93"));
-    CaseEvent caseEvent = FixtureHelper.loadPackageFixtures(CaseEvent[].class).get(0);
-    SurveyUpdate surveyUpdate = new SurveyUpdate();
-    surveyUpdate.setSurveyId("c45de4dc-3c3b-11e9-b210-d663bd873d93");
-    surveyUpdate.setSampleDefinitionUrl("test/social.json");
-    CollectionExercise collectionExercise = new CollectionExercise();
-    collectionExercise.setCollectionExerciseId(
-        caseEvent.getPayload().getCaseUpdate().getCollectionExerciseId());
-    when(mockRespondentDataRepo.readSurvey(any())).thenReturn(Optional.empty());
-    caseEvent.setHeader(header);
-    target.acceptCaseEvent(caseEvent);
-    verify(mockRespondentDataRepo, times(0)).readCollectionExercise(any());
-    verify(mockRespondentDataRepo, times(0))
-        .writeCaseUpdate(caseEvent.getPayload().getCaseUpdate());
+    CTPException thrown =
+        assertThrows(CTPException.class, () -> target.acceptCaseEvent(caseEvent));
+
+    assertEquals(CTPException.Fault.SYSTEM_ERROR, thrown.getFault());
+    assertEquals("Non Specific Error", thrown.getMessage());
```

Comment:
I don't think testing for "Non Specific Error" adds any value. I would remove this line

Reply:
Done